workflow/controller/controller.go

```diff
@@ -749,11 +759,15 @@ func (wfc *WorkflowController) processNextItem(ctx context.Context) bool {
 		woc.persistUpdates(ctx)
 		return true
 	}
-
 	startTime := time.Now()
 	woc.operate(ctx)
 	wfc.metrics.OperationCompleted(time.Since(startTime).Seconds())
 	if woc.wf.Status.Fulfilled() {
+		err = woc.DeleteAgentPod(ctx)
+		if err != nil {
+			woc.log.WithError(err).Error("unable to delete the agent pod")
```

Comment:
good -  but why not just run `woc.controller.queuePodForCleanup(woc.wf.Namespace, podName, deletePod)`